go/cmd/dolt/commands/sql.go

```diff
@@ -773,43 +781,48 @@ func execShell(sqlCtx *sql.Context, qryist cli.Queryist, format engine.PrintResu
 			query = strings.TrimSuffix(query, terminator)
 		}
 
-		var nextPrompt string
-		var multiPrompt string
+		cont := true
 		var sqlSch sql.Schema
 		var rowIter sql.RowIter
-
-		func() {
+		cont = func() bool {
 			subCtx, stop := signal.NotifyContext(initialCtx, os.Interrupt, syscall.SIGTERM)
 			defer stop()
 
 			sqlCtx := sql.NewContext(subCtx, sql.WithSession(sqlCtx.Session))
 
-			if sqlSch, rowIter, err = processQuery(sqlCtx, query, qryist); err != nil {
-				verr := formatQueryError("", err)
-				shell.Println(verr.Verbose())
-			} else if rowIter != nil {
-				switch closureFormat {
-				case engine.FormatTabular, engine.FormatVertical:
-					err = engine.PrettyPrintResultsExtended(sqlCtx, closureFormat, sqlSch, rowIter)
-				default:
-					err = engine.PrettyPrintResults(sqlCtx, closureFormat, sqlSch, rowIter)
-				}
-
+			_, foundCmd := findSlashCmd(query)
```

Comment:
I think you can use the `cli.Command` returned from this method passed into `handleSlashCommand` method so you don't have to call `findSlashCmd` twice if `foundCmd` is true.